lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -74,6 +76,116 @@ Type *getIntOrIntVectorTyForCast(LLVMContext &C, Type *Ty) {
   return IntTy;
 }
 
+Value *MemoryOrderSemantics(Value *order, bool is_global,
+                            Instruction *InsertBefore,
+                            spv::MemorySemanticsMask base_semantics) {
+  enum AtomicMemoryOrder : uint32_t {
+    kMemoryOrderRelaxed = 0,
+    kMemoryOrderAcquire = 2,
+    kMemoryOrderRelease = 3,
+    kMemoryOrderAcqRel = 4,
+    kMemoryOrderSeqCst = 5
+  };
+
+  IRBuilder<> builder(InsertBefore);
+
+  // Constants for OpenCL C 2.0 memory_order.
+  const auto relaxed = builder.getInt32(AtomicMemoryOrder::kMemoryOrderRelaxed);
+  const auto acquire = builder.getInt32(AtomicMemoryOrder::kMemoryOrderAcquire);
+  const auto release = builder.getInt32(AtomicMemoryOrder::kMemoryOrderRelease);
+  const auto acq_rel = builder.getInt32(AtomicMemoryOrder::kMemoryOrderAcqRel);
+
+  // Constants for SPIR-V ordering memory semantics.
+  const auto RelaxedSemantics = builder.getInt32(spv::MemorySemanticsMaskNone);
+  const auto AcquireSemantics =
+      builder.getInt32(spv::MemorySemanticsAcquireMask);
+  const auto ReleaseSemantics =
+      builder.getInt32(spv::MemorySemanticsReleaseMask);
+  const auto AcqRelSemantics =
+      builder.getInt32(spv::MemorySemanticsAcquireReleaseMask);
+  const auto SeqCstSemantics =
+      builder.getInt32(spv::MemorySemanticsSequentiallyConsistentMask);
+
+  // Constants for SPIR-V storage class semantics.
+  const auto UniformSemantics =
+      builder.getInt32(spv::MemorySemanticsUniformMemoryMask);
+  const auto WorkgroupSemantics =
+      builder.getInt32(spv::MemorySemanticsWorkgroupMemoryMask);
+
+  // Instead of sequentially consistent, use acquire, release or acquire
+  // release semnatics.
+  Value *base_order = SeqCstSemantics;
```

Comment:
Could just do ` = nullptr;` and remove the `SeqCstSemantics` declaration above since it's never actually used?

Unless the point was to highlight that we're replacing `SeqCst` with something else...